tests/api_resources/test_subscription.py

```diff
@@ -1,162 +1,137 @@
+        resource = stripe.Subscription.create(
+            customer='cus_123',
+            items=[{"plan": "foo", "quantity": 3}]
+        )
+        self.assert_requested(
             'post',
-            '/v1/subscriptions/test_sub',
+            '/v1/subscriptions',
             {
+                'customer': 'cus_123',
                 'items': {
                     "0": {
                         "plan": "foo",
-                        "id": "si",
+                        "quantity": 3
                     },
                 },
             },
-            None
         )
+        self.assertIsInstance(resource, stripe.Subscription)
 
-    def test_create_subscription_items(self):
-        stripe.Subscription.create(items=[{"plan": "foo", "quantity": 3}])
-        self.requestor_mock.request.assert_called_with(
+    def test_is_modifiable_with_items(self):
+        resource = stripe.Subscription.modify(
+            TEST_RESOURCE_ID,
+            items=[{"id": "si", "plan": "foo"}]
+        )
+        self.assert_requested(
             'post',
-            '/v1/subscriptions',
+            '/v1/subscriptions/%s' % TEST_RESOURCE_ID,
             {
                 'items': {
                     "0": {
                         "plan": "foo",
-                        "quantity": 3,
+                        "id": "si"
                     },
                 },
             },
-            None
-        )
-
-    def test_delete_subscription(self):
-        subscription = stripe.Subscription.construct_from({
-            'id': 'test_sub',
-            'customer': 'test_cus',
-        }, 'api_key')
-
-        subscription.delete()
-
-        self.requestor_mock.request.assert_called_with(
-            'delete',
-            '/v1/subscriptions/test_sub',
-            {},
-            None
-        )
-
-    def test_delete_subscription_discount(self):
-        subscription = stripe.Subscription.construct_from({
-            'id': 'test_sub',
-            'customer': 'test_cus',
-            'coupon': 'test_discount'
-        }, 'api_key')
-
-        subscription.delete_discount()
-
-        self.requestor_mock.request.assert_called_with(
-            'delete',
-            '/v1/subscriptions/test_sub/discount'
         )
+        self.assertIsInstance(resource, stripe.Subscription)
+
+    # TODO: Fix this test
+    # def test_is_saveable_with_items(self):
+    #    resource = stripe.Subscription.retrieve(TEST_RESOURCE_ID)
+    #    resource.items = [{"id": "si", "plan": "foo"}]
+    #    resource.save()
+    #    self.assert_requested(
+    #        'post',
+    #        '/v1/subscriptions/%s' % TEST_RESOURCE_ID,
+    #        {
+    #            'items': {
+    #                "0": {
+    #                    "plan": "foo",
+    #                    "id": "si"
+    #                },
+    #            },
+    #        },
+    #    )
+    #    self.assertIsInstance(resource, stripe.Subscription)
+
+
+    # TODO: Test serialize
```

Comment:
@ob-stripe I was not sure how to test serialize so leaving this one to you

Reply:
Ack.

Reply:
I've added two tests for `serialize` and what do you know, one of them revealed a bug! Yay tests.

The bug is identical to one I fixed in stripe-ruby a few weeks back, so hopefully it shouldn't be too hard to fix stripe-python as well. I'll tackle this tomorrow.

Reply:
I've pushed a fix in #373 and added tests there, and removed the tests from this branch for now. Once #373 is merged we can rebase and bring in the tests as well (they don't actually use stripe-mock or anything since they're just testing the `serialize` method).